xarray/backends/common.py

```diff
@@ -329,11 +329,11 @@ def encode(self, variables, attributes):
         attributes : dict-like
 
         """
-        variables = {k: self.encode_variable(v) for k, v in variables.items()}
+        variables = {k: self.encode_variable(v, k) for k, v in variables.items()}
         attributes = {k: self.encode_attribute(v) for k, v in attributes.items()}
         return variables, attributes
 
-    def encode_variable(self, v):
+    def encode_variable(self, v, name):
```

Comment:
```suggestion
    def encode_variable(self, v, name: str | None = None):
```
I'm not 100% any third party libs are using this API but making `name` optional would be a good way to avoid making this a breaking change.